libpod/image/image.go

```diff
@@ -142,6 +142,25 @@ func (ir *Runtime) NewFromLocal(name string) (*Image, error) {
 	return ir.newImage(updatedInputName, localImage), nil
 }
 
+// matchesPlatform returns true if the image matches the specified os, arch and
+// variant.  Note that os, arch and variant are ignored if they are empty.
+func (i *Image) matchesPlatform(ctx context.Context, os, arch, variant string) (bool, error) {
+	data, err := i.inspect(ctx, false)
+	if err != nil {
+		return false, errors.Wrap(err, "error determining image's platform")
+	}
+	if os != "" && data.Os != os {
+		return false, nil
+	}
+	if arch != "" && data.Architecture != arch {
+		return false, nil
+	}
+	if variant != "" && data.Variant != arch {
```

Comment:
Comparing `data.Variant` with `arch` instead of `variant` looks wrong.

Reply:
Thanks for catching!